src/main/java/seedu/address/model/person/Person.java

```diff
@@ -130,6 +139,38 @@ public List<Interaction> addInteractions(List<Interaction> interactions) {
         return this.interactions;
     }
 
+    public Optional<LocalDate> getFollowUpDate() {
+        if (interactions.isEmpty() || lead == null) {
+            return Optional.empty();
+        }
+        LocalDate latestInteractionDate = interactions.get(interactions.size() - 1).getDate();
+        //TODO: Not sure if the lastest interaction is always the last one in the list
+        int weeksToAdd = lead.getFollowUpPeriod();
+        return Optional.of(latestInteractionDate.plusWeeks(weeksToAdd));
+    }
+
+    /**
+     * Adds a Reminder to the list.
+     * The Reminder must not already exist in the list.
+     */
+    public void updateReminder() {
+        //Person must have lead and interaciton to have a reminder
+        //If person has no lead or interaction, remove reminder
+        if (this.getFollowUpDate().isEmpty()) {
+            this.reminder = Optional.empty();
```

Comment:
Do we need to do this?

Reply:
This is if a persons interactions get deleted for example, then we gotta remove his interactions. Unless you mean something else